### PR TITLE
Added some clear error messages and fixed stdResidsCovTab issue

### PR DESCRIPTION
**Pull request by LSLindeloo**
fixes https://github.com/jasp-stats/jasp-test-release/issues/2008 
fixes https://github.com/jasp-stats/jasp-test-release/issues/2009
fixes https://github.com/jasp-stats/jasp-test-release/issues/2010
fixes https://github.com/jasp-stats/jasp-test-release/issues/2011

### CODE REVIEW

**Comment by LSLindeloo** [comment]
The reason why the standardized residual covariance matrix cannot be computed for se method 'bootstrap', is because `lavaan::lav_model_h1_acov()` doesn't return the Asymptotic Covariance Matrix if the se method is set to 'bootstrap':

`# which type of se?
if(lavoptions$se %in% c("standard", "none")) {
    ACOV[[g]] <- 1/Ng * I1.g.inv
} else if(lavoptions$se %in% c("robust.huber.white", "robust.sem")) {
    ACOV[[g]] <- 1/Ng * (I1.g.inv %*% J1[[g]] %*% I1.g.inv)
}
}

ACOV
}`

Do you have any idea why this could be the case? 
I tried to find out how to compute the ACOV in case of se method = 'bootstrap', but untill now I haven't found a clear answer.

File: R/sem.R [hunk 1]
```diff
@@ -487,12 +491,15 @@ checkLavaanModel <- function(model, availableVars) {
...
- if(options$missing == "listwise"){
-   nrm <- nrow(dataset) - lavaan::lavInspect(semResults[[1]], "ntotal")
-   missingFootnote <- gettextf("A total of %g cases were removed due to missing values. You can avoid this by choosing 'FIML' under 'Missing Data Handling' in the Estimation options.",
-                               nrm)
-   fittab$addFootnote(message = missingFootnote)
- }
+  if(options$missing == "listwise") {
+    nrm <- nrow(dataset) - lavaan::lavInspect(semResults[[1]], "ntotal")
+    if (nrm != 0) {
+      missingFootnote <- gettextf("A total of %g cases were removed due to missing values. You can avoid this by choosing 'FIML' under 'Missing Data Handling' in the Estimation options.",
```
Comment (Kucharssim) [inline R/sem.R:497]: this is probably fine, but `%s` would be probably sufficient, or perhaps `%i` if you want to insist that it's an integer
```suggestion
      missingFootnote <- gettextf("A total of %g cases were removed due to missing values. You can avoid this by choosing 'FIML' under 'Missing Data Handling' in the Estimation options.",
```

File: R/sem.R [hunk 2]
```diff
@@ -989,17 +1000,21 @@ checkLavaanModel <- function(model, availableVars) {
...
-    fitic$addColumnInfo(name = "value", title = gettext("Value"), type = "number", format = "sf:4;dp:3")
-  } else {
-    for (i in seq_along(options[["models"]])) {
-      fitic$addColumnInfo(name = paste0("value_", i), title = options[["models"]][[i]][["modelName"]], type = "number",
-                          format = "sf:4;dp:3")
+  if (!options$estimator %in% c("DWLS", "GLS", "WLS", "ULS")) {
+    fitms[["incrits"]] <- fitic <- createJaspTable(gettext("Information criteria"))
+    fitic$addColumnInfo(name = "index", title = "",               type = "string")
+    if (length(options[["models"]]) < 2) {
+      fitic$addColumnInfo(name = "value", title = gettext("Value"), type = "number", format = "sf:4;dp:3")
```
Comment (Kucharssim) [inline R/sem.R:1007]: I am not sure why this formatting was added in the first place, but it seems redundant (you can remove it for the other columns as well).
```suggestion
      fitic$addColumnInfo(name = "value", title = gettext("Value"), type = "number")
```

File: R/sem.R [hunk 3]
```diff
@@ -1386,15 +1404,23 @@ checkLavaanModel <- function(model, availableVars) {
...
       # actually compute the implied covariance
-      sv <- lavaan::residuals(fit, type = "standardized")
-      sr <- sv$cov
-      sr[upper.tri(sr)] <- NA
-
-      for (i in 1:ncol(sr)) {
-        nm <- colnames(sr)[i]
-        srtab$addColumnInfo(nm, title = .unv(nm), type = "number", format = "sf:4;dp:3;p:.001")
+      if (options[["se"]] == "bootstrap") {
+        srtab$setError(gettext("The standardized residual covariance table is currently unavailable when the error calculation method is 'Bootstrap'"))
```
Comment (Kucharssim) [inline R/sem.R:1408]: alternatively you can just ignore this option and disable the standardized residuals option in QML when `bootstrap` is selected. But this is a fine solution as it is.

**Review by Kucharssim — CHANGES_REQUESTED** [review]
good work, all issues are fixed! Just little nitpicks

**Comment by boutinb** [comment]
@LSLindeloo Would it be possible to correct today this fix according to the remarks of Simon? It would be nice to be able to have a release this week.

**Review by Kucharssim — APPROVED** [review]

**Comment by LSLindeloo** [comment]
Yes, of course!  Will do it right away!

Op ma 26 sep. 2022 om 12:51 schreef bruno boutin ***@***.***>:

**Pull request merged**